sdks/java/harness/src/main/java/org/apache/beam/fn/harness/BeamFnDataReadRunner.java

```diff
@@ -22,7 +22,7 @@
 
 import com.google.auto.service.AutoService;
 import com.google.common.collect.ImmutableMap;
-import com.google.common.collect.Multimap;
+import com.google.common.collect.ListMultimap;
```

Comment:
Yes, we want to ensure we allow for duplicates and to prevent the erroneous usage of SetMultimap/HashMultimap.

Flatten may consume the same PCollection multiple times.

As you put it, we are clarifying the contract to ensure people know that duplicates are allowed.